utility/block.go

```diff
@@ -93,31 +97,41 @@ func (u *UtilityContext) ApplyBlock() (appHash []byte, err error) {
 		if err := tx.ValidateBasic(); err != nil {
 			return nil, err
 		}
+		// DISCUSS(#315): Currently, the pattern is allowing nil err with an error transaction...
```

Comment:
Issue is closed. should we open another?

Reply:
I'm going to create a ticket explicitly for removing `TxResult` from the interface

Reply:
#346